tests/cupy_tests/random_tests/test_generator_api.py

```diff
@@ -194,36 +194,52 @@ def test_invalid_dtypes(self):
 @testing.gpu
 @testing.fix_random()
 class TestIntegers(GeneratorTestCase):
-    # TODO(niboshi):
-    #   Test soundness of distribution.
-    #   Currently only reprocibility is checked.
-
     target_method = 'integers'
 
-    def test_randint_1(self):
+    def test_integers_1(self):
         self.generate(3)
 
-    def test_randint_2(self):
+    def test_integers_2(self):
         self.generate(3, 4, size=(3, 2))
 
-    def test_randint_empty1(self):
+    def test_integers_empty1(self):
         self.generate(3, 10, size=0)
 
-    def test_randint_empty2(self):
+    def test_integers_empty2(self):
         self.generate(3, size=(4, 0, 5))
 
-    def test_randint_overflow(self):
+    def test_integers_overflow(self):
         self.generate(numpy.int8(-100), numpy.int8(100))
 
-    def test_randint_float1(self):
+    def test_integers_float1(self):
         self.generate(-1.2, 3.4, 5)
 
-    def test_randint_float2(self):
+    def test_integers_float2(self):
         self.generate(6.7, size=(2, 3))
 
-    def test_randint_int64_1(self):
+    def test_integers_int64_1(self):
         self.generate(2**34, 2**40, 3)
 
+    @_condition.repeat_with_success_at_least(10, 3)
+    def test_integers_ks(self):
+        self.check_ks(0.05)(
+            low=100, high=1000, size=2000)
+
+    @_condition.repeat_with_success_at_least(10, 3)
+    def test_integers_ks_low(self):
+        self.check_ks(0.05)(
+            low=100, size=2000)
```

Comment:
For discrete distributions, K-S test might be unusual and `check_ks` returns inaccurate p-value.  However, adding a distribution test is a great progress and I hope the bin size (100) is sufficiently large.

Reply:
yes, I actually thought of converting the distribution results to the`[0, 1)` interval for the test but I wasn't sure of it.